homeassistant/components/renault/renault_entities.py

```diff
@@ -1,22 +1,27 @@
 """Base classes for Renault entities."""
 from __future__ import annotations
 
-from typing import Any, Generic, Optional, TypeVar
+from collections.abc import Mapping
+from dataclasses import dataclass
+from typing import Any, Generic, Optional, TypeVar, cast
 
-from renault_api.kamereon.enums import ChargeState, PlugState
-from renault_api.kamereon.models import (
-    KamereonVehicleBatteryStatusData,
-    KamereonVehicleChargeModeData,
-    KamereonVehicleCockpitData,
-    KamereonVehicleHvacStatusData,
-)
-
-from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity import Entity, EntityDescription
+from homeassistant.helpers.typing import StateType
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
-from homeassistant.util import slugify
 
 from .renault_vehicle import RenaultVehicleProxy
 
+
+@dataclass
+class RenaultEntityDescription(EntityDescription):
+    """Class describing Renault entities."""
+
+    coordinator: str = ""
+    data_key: str = ""
+    entity_class: str = ""
```

Comment:
Those keys seem to be required. It would be better to create a mixin class for them. Take a look how it's done here as an example: https://github.com/home-assistant/core/blob/8d3ccad38ead4aab6524b686353b65616f6aa0bc/homeassistant/components/melcloud/sensor.py#L24-L36

Reply:
This should be OK now